test/parallel/test-process-no-deprecation.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+// Flags: --expose_internals --no-warnings
+
+// The --no-warnings flag only supresses writing the warning to stderr, not the
+// emission of the corresponding event. This test file can be run without it.
+
+process.noDeprecation = true;
+
+const common = require('../common');
+const assert = require('assert');
+
+function listener(warning) {
+  assert.notStrictEqual(warning.name, 'DeprecationWarning');
```

Comment:
If I am not wrong this function should not be called right? I think `assert(false, 'this should not be called');` or something like this would be better I guess.